src/core/qfieldcloudprojectsmodel.h

```diff
@@ -487,6 +491,8 @@ class QFieldCloudProjectsModel : public QAbstractListModel
     QString getJobTypeAsString( JobType jobType ) const;
 
     void downloadFileConnections( const QString &projectId, const QString &fileName );
+    void insertProjectList( const QList<CloudProject *> &projects );
+    bool isProjectExists( CloudProject *project );
```

Comment:
I don't think you'll need that here, just make sure that the insertProjects function checks for duplicates.